
From: David Howells <dhowells@redhat.com>

The attFrom: David Howells <dhowells@redhat.com>
ached patch adds the new error codes I added for key-related errors to
those archs that don't make use of <asm-generic/errno.h>, including Alpha,
MIPS, PA-RISC, Sparc and Sparc64. This is required to compile with CONFIG_KEYS
on those platforms.

Signed-Off-By: David Howells <dhowells@redhat.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/include/asm-alpha/errno.h   |    4 ++++
 25-akpm/include/asm-mips/errno.h    |    4 ++++
 25-akpm/include/asm-parisc/errno.h  |    4 ++++
 25-akpm/include/asm-sparc/errno.h   |    4 ++++
 25-akpm/include/asm-sparc64/errno.h |    4 ++++
 5 files changed, 20 insertions(+)

diff -puN include/asm-alpha/errno.h~keys-new-error-codes-for-alpha-mips-pa-risc-sparc-sparc64 include/asm-alpha/errno.h
--- 25/include/asm-alpha/errno.h~keys-new-error-codes-for-alpha-mips-pa-risc-sparc-sparc64	Tue Aug 31 14:36:27 2004
+++ 25-akpm/include/asm-alpha/errno.h	Tue Aug 31 14:36:27 2004
@@ -110,5 +110,9 @@
 
 #define ENOMEDIUM	129	/* No medium found */
 #define EMEDIUMTYPE	130	/* Wrong medium type */
+#define	ENOKEY		131	/* Required key not available */
+#define	EKEYEXPIRED	132	/* Key has expired */
+#define	EKEYREVOKED	133	/* Key has been revoked */
+#define	EKEYREJECTED	134	/* Key was rejected by service */
 
 #endif
diff -puN include/asm-mips/errno.h~keys-new-error-codes-for-alpha-mips-pa-risc-sparc-sparc64 include/asm-mips/errno.h
--- 25/include/asm-mips/errno.h~keys-new-error-codes-for-alpha-mips-pa-risc-sparc-sparc64	Tue Aug 31 14:36:27 2004
+++ 25-akpm/include/asm-mips/errno.h	Tue Aug 31 14:36:27 2004
@@ -110,6 +110,10 @@
  */
 #define ENOMEDIUM	159	/* No medium found */
 #define EMEDIUMTYPE	160	/* Wrong medium type */
+#define	ENOKEY		161	/* Required key not available */
+#define	EKEYEXPIRED	162	/* Key has expired */
+#define	EKEYREVOKED	163	/* Key has been revoked */
+#define	EKEYREJECTED	164	/* Key was rejected by service */
 
 #define EDQUOT		1133	/* Quota exceeded */
 
diff -puN include/asm-parisc/errno.h~keys-new-error-codes-for-alpha-mips-pa-risc-sparc-sparc64 include/asm-parisc/errno.h
--- 25/include/asm-parisc/errno.h~keys-new-error-codes-for-alpha-mips-pa-risc-sparc-sparc64	Tue Aug 31 14:36:27 2004
+++ 25-akpm/include/asm-parisc/errno.h	Tue Aug 31 14:36:27 2004
@@ -67,6 +67,10 @@
 #define	EREMOTEIO	181	/* Remote I/O error */
 #define	ENOMEDIUM	182	/* No medium found */
 #define	EMEDIUMTYPE	183	/* Wrong medium type */
+#define	ENOKEY		184	/* Required key not available */
+#define	EKEYEXPIRED	185	/* Key has expired */
+#define	EKEYREVOKED	186	/* Key has been revoked */
+#define	EKEYREJECTED	187	/* Key was rejected by service */
 
 /* We now return you to your regularly scheduled HPUX. */
 
diff -puN include/asm-sparc64/errno.h~keys-new-error-codes-for-alpha-mips-pa-risc-sparc-sparc64 include/asm-sparc64/errno.h
--- 25/include/asm-sparc64/errno.h~keys-new-error-codes-for-alpha-mips-pa-risc-sparc-sparc64	Tue Aug 31 14:36:27 2004
+++ 25-akpm/include/asm-sparc64/errno.h	Tue Aug 31 14:36:27 2004
@@ -101,5 +101,9 @@
 
 #define ENOMEDIUM       125     /* No medium found */
 #define EMEDIUMTYPE     126     /* Wrong medium type */
+#define	ENOKEY		127	/* Required key not available */
+#define	EKEYEXPIRED	128	/* Key has expired */
+#define	EKEYREVOKED	129	/* Key has been revoked */
+#define	EKEYREJECTED	130	/* Key was rejected by service */
 
 #endif /* !(_SPARC64_ERRNO_H) */
diff -puN include/asm-sparc/errno.h~keys-new-error-codes-for-alpha-mips-pa-risc-sparc-sparc64 include/asm-sparc/errno.h
--- 25/include/asm-sparc/errno.h~keys-new-error-codes-for-alpha-mips-pa-risc-sparc-sparc64	Tue Aug 31 14:36:27 2004
+++ 25-akpm/include/asm-sparc/errno.h	Tue Aug 31 14:36:27 2004
@@ -101,5 +101,9 @@
 
 #define	ENOMEDIUM	125	/* No medium found */
 #define	EMEDIUMTYPE	126	/* Wrong medium type */
+#define	ENOKEY		127	/* Required key not available */
+#define	EKEYEXPIRED	128	/* Key has expired */
+#define	EKEYREVOKED	129	/* Key has been revoked */
+#define	EKEYREJECTED	130	/* Key was rejected by service */
 
 #endif
_
